cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -301,44 +301,48 @@ func performEtcdStaticPodUpgrade(waiter apiclient.Waiter, pathMgr StaticPodPathM
 		return true, fmt.Errorf("fatal error when trying to upgrade the etcd cluster: %v, rolled the state back to pre-upgrade state", err)
 	}
 
-	fmt.Println("[upgrade/etcd] waiting for etcd pod to become available after TLS upgrade")
-
-	// Initialize the new etcd client if it wasn't pre-initialized
-	if newEtcdClient == nil {
-		client, err := etcdutil.NewStaticPodClient(
-			[]string{"localhost:2379"},
-			constants.GetStaticPodDirectory(),
-			cfg.CertificatesDir,
-		)
-		if err != nil {
-			return true, fmt.Errorf("fatal error creating etcd client: %v", err)
-		}
-		newEtcdClient = client
-	}
+	if isTLSUpgrade {
+		fmt.Println("[upgrade/etcd] waiting for etcd pod to become available after TLS upgrade")
```

Comment:
If we diff this against master, I believe the L7 etcd check is always supposed to run regardless of TLS Upgrade.
The only difference would be we now have a waiter for extra safety.

fmt.Println("[upgrade/etcd] waiting for etcd to become healthy")

We could consider setting the initial delay only if isTLSUpgrade though since it's to avoid a race specific to not waiting for podHash

Reply:
Ah, indeed. I was wondering why I had originally left the conditional out :joy:

Reply:
Latest commit has removed the conditional.